processor/processor.go

```diff
@@ -700,6 +700,33 @@ func makeCommonMetadataFromSingularEvent(singularEvent types.SingularEventT, bat
 	return &commonMetadata
 }
 
+func makeCommonMetadataFromSingularEventWithoutMarshalling(singularEvent types.SingularEventT, batchEvent *jobsdb.JobT, receivedAt time.Time, source backendconfig.SourceT) *transformer.MetadataT {
+	commonMetadata := transformer.MetadataT{}
+	commonMetadata.SourceID = gjson.GetBytes(batchEvent.Parameters, "source_id").Str
+	commonMetadata.WorkspaceID = source.WorkspaceID
+	commonMetadata.Namespace = config.GetKubeNamespace()
+	commonMetadata.InstanceID = config.GetInstanceID()
+	commonMetadata.RudderID = batchEvent.UserID
+	commonMetadata.JobID = batchEvent.JobID
+	commonMetadata.MessageID = misc.GetStringifiedData(singularEvent["messageId"])
+	commonMetadata.ReceivedAt = receivedAt.Format(misc.RFC3339Milli)
+	commonMetadata.SourceType = source.SourceDefinition.Name
+	commonMetadata.SourceCategory = source.SourceDefinition.Category
+
+	commonMetadata.SourceBatchID = misc.MapLookup(singularEvent, "", "context", "sources", "batch_id").(string)
```

Comment:
all `.(string)` type assertions below can theoretically fail. With this small trick you are safe and you no longer need to pass "" as a default value to `MapLookup` (can remove the parameter and return `nil` instead)
```suggestion
	commonMetadata.SourceBatchID, _ = misc.MapLookup(singularEvent, "", "context", "sources", "batch_id").(string)
```

Reply:
Nice. Something felt amiss. This makes sense.